pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -836,7 +836,7 @@ func (m *ManagerImpl) allocateContainerResources(pod *v1.Pod, container *v1.Cont
 	for k, v := range container.Resources.Limits {
 		resource := string(k)
 		needed := int(v.Value())
-		klog.V(3).InfoS("Looking for needed resources", "needed", needed, "resourceName", resource)
+		klog.V(3).InfoS("Looking for needed resources", "resourceName", resource, "pod", klog.KObj(pod), "containerName", container.Name, "needed", needed)
```

Comment:
worth recording in the commit message that we use podUID only when it is the only identifying info we have; if we have the full pod object, we prefer KObj and we defer to the contextual logging change the final decision about what to log, because always logging both can be redundant.

Reply:
Yeah, good point. I am happy to update the commit message if you would like.

Reply:
If for whatever reason you need to resubmit, please do. Otherwise, let's move on. This PR is LGTM+approved and in the merge pool, so let's go